docker-api/src/main/java/com/yahoo/vespa/hosted/dockerapi/metrics/MetricReceiverWrapper.java

```diff
@@ -68,40 +76,60 @@ public GaugeWrapper declareGauge(String application, Dimensions dimensions, Stri
         }
     }
 
-    public List<DimensionMetrics> getAllMetrics() {
+    public List<DimensionMetrics> getDefaultMetrics() {
+        return getMetricsByType(DimensionType.DEFAULT);
+    }
+
+    // For testing, returns same as getDefaultMetrics(), but without "timestamp"
+    public Set<Map<String, Object>> getDefaultMetricsRaw() {
         synchronized (monitor) {
-            List<DimensionMetrics> dimensionMetrics = new ArrayList<>();
-            applicationMetrics.forEach((application, applicationMetrics) -> applicationMetrics.metricsByDimensions().entrySet().stream()
+            Set<Map<String, Object>> dimensionMetrics = new HashSet<>();
+            metrics.getOrDefault(DimensionType.DEFAULT, new HashMap<>())
+                    .forEach((application, applicationMetrics) -> applicationMetrics.metricsByDimensions().entrySet().stream()
                     .map(entry -> new DimensionMetrics(application, entry.getKey(),
                             entry.getValue().entrySet().stream().collect(Collectors.toMap(Map.Entry::getKey, value -> value.getValue().getValue()))))
+                    .map(DimensionMetrics::getMetrics)
                     .forEach(dimensionMetrics::add));
             return dimensionMetrics;
         }
     }
 
-    // For testing, returns same as getAllMetrics(), but without "timestamp"
-    public Set<Map<String, Object>> getAllMetricsRaw() {
+    public List<DimensionMetrics> getMetricsByType(DimensionType type) {
         synchronized (monitor) {
-            Set<Map<String, Object>> dimensionMetrics = new HashSet<>();
-            applicationMetrics.forEach((application, applicationMetrics) -> applicationMetrics.metricsByDimensions().entrySet().stream()
+            List<DimensionMetrics> dimensionMetrics = new ArrayList<>();
+            metrics.getOrDefault(type, new HashMap<>())
+                    .forEach((application, applicationMetrics) -> applicationMetrics.metricsByDimensions().entrySet().stream()
                     .map(entry -> new DimensionMetrics(application, entry.getKey(),
                             entry.getValue().entrySet().stream().collect(Collectors.toMap(Map.Entry::getKey, value -> value.getValue().getValue()))))
-                    .map(DimensionMetrics::getMetrics)
                     .forEach(dimensionMetrics::add));
             return dimensionMetrics;
         }
     }
 
+    public void deleteMetricByDimension(String name,  Dimensions dimensionsToRemove, DimensionType type) {
+        try{
+            metrics.get(type)
+                    .get(name)
+                    .metricsByDimensions()
+                    .remove(dimensionsToRemove);
+        } catch (NullPointerException e) {}
```

Comment:
`NullPointerException` is too generic and could hide an actual problem with the code, something like
```
Optional.ofNullable(metrics.get(type))
        .map(m -> m.get(name))
        .map(ApplicationMetrics::metricsByDimensions)
        .ifPresent(m -> m.remove(dimensionsToRemove));
```
would be better